Triangulation_2/doc/Triangulation_2/Concepts/ConstrainedTriangulationTraits_2.h

```diff
@@ -39,7 +39,7 @@ class ConstrainedTriangulationTraits_2 {
 /*!
 A function object whose `operator()` computes the intersection of two segments.
 
-`Object_2 operator()(Segment_2 s1, Segment_2 s2);`
+`boost:optional<boost::variant<Point_2,Segment_2> > operator()(Segment_2 s1, Segment_2 s2);`
```

Comment:
We do not want to document an API with `boost::optional` and `boost::variant`, or it will break again next time we change the API (for `std::optional` and `std::variant` for example). We will have to find a different way to document that.

Reply:
What are our options
* With C++17 the `std` versions of `optional`   and `variant` arrive.  Do you want `CGAL::cpp17` abstractions?   
* Or ` CGAL::cpp11::result_of<Kernel::Intersect_2(A, B)>::type`?   
* Even with `auto`  we at have to document the API   for getting the result out of it.
* Yet another alternative is to have an API that is fully controlled by us, e.g. with a reference to a point and a segment, and an `enum` telling us if there is no intersection or an intersection point or segment.  To stick to `Object` falls in the same category.  However, with this alternative we do not  achieve the goal that a kernel is a concept for the traits class.